### PR TITLE
JAMES-1698 Allow to configure JMAP port into jmap.properties file

### PR DESCRIPTION
**Pull request by rouazana**
JMAP port defaults to 80.
If I want to use an other port, I must modify DefaultPortConfiguration class.
This is not very admin-friendly.

So this commit removes java port binding and use configuration instead.

### CODE REVIEW

File: server/container/cassandra-guice/src/main/java/org/apache/james/jmap/JMAPModule.java [hunk 1]
```diff
@@ -52,6 +51,7 @@ JMAPConfiguration provideConfiguration(FileSystem fileSystem) throws Configurati
                 .keystore(configuration.getString("tls.keystoreURL"))
                 .secret(configuration.getString("tls.secret"))
                 .jwtPublicKeyPem(loadPublicKey(fileSystem, Optional.ofNullable(configuration.getString("jwt.publickeypem.url"))))
+                .port(Optional.ofNullable(configuration.getInteger("jmap.port", DEFAULT_JMAP_PORT)))
```
Comment (aduprat) [inline server/container/cassandra-guice/src/main/java/org/apache/james/jmap/JMAPModule.java]: I would use `Optional.of` method, i don't think we want to manage random port at runtime.

Comment (rouazana) [inline server/container/cassandra-guice/src/main/java/org/apache/james/jmap/JMAPModule.java]: Yep, I was about to fixup that.

**Comment by fvn-linagora** [comment]
:+1:

**Comment by aduprat** [comment]
:+1:

**Pull request merged**